src/protocol/context/mod.rs

```diff
@@ -89,26 +89,11 @@ mod tests {
     use super::*;
     use crate::test_fixture::{Reconstruct, Runner, TestWorld, TestWorldConfig};
 
-    trait AsReplicated<F: Field> {
-        fn left(&self) -> F;
-        fn right(&self) -> F;
-    }
-
-    impl<F: Field> AsReplicated<F> for Replicated<F> {
-        fn left(&self) -> F {
-            (self as &Replicated<F>).left()
-        }
-
-        fn right(&self) -> F {
-            (self as &Replicated<F>).right()
-        }
-    }
-
     /// This looks weird because it uses `MaliciousReplicated::rx()` value instead of `x`.
     /// Malicious context intentionally disallows access to `x` without validating first and
     /// here it does not matter at all. It needs just some value to send (any value would do just
     /// fine)
-    impl<F: Field> AsReplicated<F> for MaliciousReplicated<F> {
+    impl<F: Field> ReplicatedSecretSharing<F> for MaliciousReplicated<F> {
         fn left(&self) -> F {
             (self as &MaliciousReplicated<F>).rx().left()
         }
```

Comment:
Wait a minute...

This looks like it is ONLY accessing `rx` and never `x`...

...I suspect that this "reshare" isn't even secure in a semi-honest sense, because if the `x` value isn't getting shuffled too, then you can still trace where records went...  or am I missing something?